### PR TITLE
Update ICU info

### PR DESCRIPTION
**Pull request by gewarren**
Related to dotnet/docs#18423.

*Hide whitespace changes in diff.*

[Preview link 1](https://review.docs.microsoft.com/en-us/dotnet/api/system.string?view=net-5.0&branch=pr-en-us-5083#strings-and-the-unicode-standard).
[Preview link 2](https://review.docs.microsoft.com/en-us/dotnet/api/system.string?view=net-5.0&branch=pr-en-us-5083#ordinal-vs-culture-sensitive-operations).

### CODE REVIEW

File: xml/System/String.xml [hunk 1]
```diff
@@ -179,7 +179,7 @@ For more information about Unicode, UTF-16, code units, code points, and the <xr
...
+ In addition, .NET supports string comparison and sorting based on the Unicode standard. .NET Framework 4 and previous versions maintain their own table of string data. This is also true of versions of .NET Framework starting with .NET Framework 4.5 running on Windows 7. Starting with .NET Framework 4.5 running on Window 8 and later versions of the Windows operating system, the runtime delegates string comparison and sorting operations to the operating system. On .NET Core and .NET 5+, string comparison and sorting information is provided by [International Components for Unicode](http://site.icu-project.org/) libraries. The following table lists the versions of .NET and the versions of the Unicode Standard on which character comparison and sorting are based.
```
Comment (safern) [inline xml/System/String.xml:182]:
```suggestion
 In addition, .NET supports string comparison and sorting based on the Unicode standard. .NET Framework 4 and previous versions maintain their own table of string data. This is also true of versions of .NET Framework starting with .NET Framework 4.5 running on Windows 7. Starting with .NET Framework 4.5 running on Windows 8 and later versions of the Windows operating system, the runtime delegates string comparison and sorting operations to the operating system. On .NET Core and .NET 5+, string comparison and sorting information is provided by [International Components for Unicode](http://site.icu-project.org/) libraries. The following table lists the versions of .NET and the versions of the Unicode Standard on which character comparison and sorting are based.
```

This is true only for Windows 19H1+ should we mention that?

File: xml/System/String.xml [hunk 2]
```diff
@@ -284,9 +284,12 @@ For more information about Unicode, UTF-16, code units, code points, and the <xr
...
 
-You can download the [Sorting Weight Tables](https://www.microsoft.com/download/details.aspx?id=10921), a set of text files that contain information on the character weights used in sorting and comparison operations for Windows operating systems, and the [Default Unicode Collation Element Table](https://www.unicode.org/Public/UCA/latest/allkeys.txt), the sort weight table for Linux and macOS.
+You can download the sorting weight tables, a set of text files that contain information on the character weights used in sorting and comparison operations, from the following links:
+
+- Windows (.NET Framework and .NET Core): [Sorting Weight Tables](https://www.microsoft.com/download/details.aspx?id=10921)
+- Windows 19H1 or later (.NET 5+) and Linux and macOS (.NET Core and .NET 5+): [Default Unicode Collation Element Table](https://www.unicode.org/Public/UCA/latest/allkeys.txt)
```
Comment (safern) [inline xml/System/String.xml:292]: I believe in other docs we reference this as `Windows 10 May 2019 Update`

File: xml/System/String.xml [hunk 3]
```diff
@@ -563,8 +566,8 @@ You can download the [Sorting Weight Tables](https://www.microsoft.com/download/
 
  ]]></format>
     </remarks>
-    <related type="ExternalDocumentation" href="https://www.microsoft.com/download/details.aspx?id=10921">Sorting Weight Tables for Windows operating systems</related>
-    <related type="ExternalDocumentation" href="https://www.unicode.org/Public/UCA/latest/allkeys.txt">Default Unicode Collation Element Table, for Linux and macOS</related>
+    <related type="ExternalDocumentation" href="https://www.microsoft.com/download/details.aspx?id=10921">Sorting Weight Tables, for Windows operating systems (.NET Framework and .NET Core only)</related>
+    <related type="ExternalDocumentation" href="https://www.unicode.org/Public/UCA/latest/allkeys.txt">Default Unicode Collation Element Table, for .NET 5+ on Linux, macOS, or Windows 19H1 or later, and .NET Core on Linux and macOS</related>
```
Comment (safern) [inline xml/System/String.xml:570]: Same note about `Windows 19H1` -> `Windows 10 May 2019 Update`.

**Review by safern — APPROVED** [review]
cc: @tarekgh @GrabYourPitchforks

File: xml/System/String.xml (hunk 3 above)
Comment (gewarren) [reply xml/System/String.xml:570]: This link text was too long so I just ended up shortening them both.

**Pull request merged**